File: app/src/main/java/com/odnovolov/randomuser/network/model/RandomUserResponse.java
package com.odnovolov.randomuser.network.model;

import java.util.*;

public class RandomUserResponse {
    Info info;
    List<Result> results;

    public static class Info {
        int page;
        int results;
        String seed;
        String version;
    }

    public static class Result {
        String cell;
        Dob dob;
        String email;
        String gender;
        Id id;
        Location location;
        Login login;
        Name name;
        String nat;
        String phone;
        Picture picture;
        Registered registered;
    }

    public static class Dob {
        int age;
        String date;
    }

    public static class Location {
        String city;
        Coordinates coordinates;
        String country;
        String postcode;
        String state;
        Street street;
        Timezone timezone;
    }

    public static class Login {
        String md5;
        String password;
        String salt;
        String sha1;
        String sha256;
        String username;
        String uuid;
    }

    public static class Name {
        String first;
        String last;
        String title;
    }

    public static class Picture {
        String large;
        String medium;
        String thumbnail;
    }

    public static class Registered {
        int age;
        String date;
    }

    public static class Street {
        String name;
        int number;
    }

    public static class Timezone {
        String description;
        String offset;
    }

    public static class Id {
        String name;
        Object value;
    }

    public static class Coordinates {
        String latitude;
        String longitude;
    }
}

File: app/src/main/java/com/odnovolov/randomuser/network/model/RandomUserResponse.kt
package com.odnovolov.randomuser.network.model

import com.odnovolov.randomuser.model.UserImpl
import java.util.*

data class RandomUserResponse(
    val info: Info,
    val results: List<Result>
) {
    fun toUsers(): List<UserImpl> {
        return results.map { result: Result ->
            UserImpl(
                id = UUID.randomUUID().toString(),
                firstName = result.name.first,
                lastName = result.name.last,
                photoUrl = result.picture.large,
                age = result.dob.age,
                city = result.location.city,
                email = result.email
            )
        }
    }

    data class Info(
        val page: Int,
        val results: Int,
        val seed: String,
        val version: String
    )

    data class Result(
        val cell: String,
        val dob: Dob,
        val email: String,
        val gender: String,
        val id: Id,
        val location: Location,
        val login: Login,
        val name: Name,
        val nat: String,
        val phone: String,
        val picture: Picture,
        val registered: Registered
    )

    data class Dob(
        val age: Int,
        val date: String
    )

    data class Location(
        val city: String,
        val coordinates: Coordinates,
        val country: String,
        val postcode: String,
        val state: String,
        val street: Street,
        val timezone: Timezone
    )

    data class Login(
        val md5: String,
        val password: String,
        val salt: String,
        val sha1: String,
        val sha256: String,
        val username: String,
        val uuid: String
    )

    data class Name(
        val first: String,
        val last: String,
        val title: String
    )

    data class Picture(
        val large: String,
        val medium: String,
        val thumbnail: String
    )

    data class Registered(
        val age: Int,
        val date: String
    )

    data class Street(
        val name: String,
        val number: Int
    )

    data class Timezone(
        val description: String,
        val offset: String
    )

    data class Id(
        val name: String,
        val value: Any
    )

    data class Coordinates(
        val latitude: String,
        val longitude: String
    )
}